bundles/org.openhab.binding.broadlink/src/main/java/org/openhab/binding/broadlink/internal/handler/BroadlinkBaseThingHandler.java

```diff
@@ -0,0 +1,293 @@
+            this.deviceKey = BroadlinkProtocol.getDeviceKey(decryptResponse);
+
+            // Update the properties, so that these values can be seen in the UI:
+            Map<String, String> properties = editProperties();
+            properties.put("id", HexUtils.bytesToHex(deviceId));
+            properties.put("key", HexUtils.bytesToHex(deviceKey));
+            updateProperties(properties);
+            logger.debug("Authenticated with id '{}' and key '{}'", HexUtils.bytesToHex(deviceId),
+                    HexUtils.bytesToHex(deviceKey));
+            authenticated = true;
+            return;
+        } catch (Exception e) {
+            throw new BroadlinkAuthenticationException("Authentication failed:" + e.getMessage(), e);
+        }
+    }
+
+    protected byte @Nullable [] sendAndReceiveDatagram(byte message[], String purpose) {
+        if (socket != null) {
+            return socket.sendAndReceive(message, purpose);
+        } else {
+            return null;
+        }
+    }
+
+    protected byte[] buildMessage(byte command, byte payload[]) throws IOException {
+        return buildMessage(command, payload, thingConfig.getDeviceType());
+    }
+
+    private byte[] buildMessage(byte command, byte payload[], int deviceType) throws IOException {
+        count = count + 1 & 0xffff;
+
+        NetworkTrafficObserver networkTrafficObserver = this.networkTrafficObserver;
+
+        if (networkTrafficObserver != null) {
+            networkTrafficObserver.onCommandSent(command);
+            networkTrafficObserver.onBytesSent(payload);
+        }
+
+        return BroadlinkProtocol.buildMessage(command, payload, count, thingConfig.getMacAddress(), deviceId,
+                HexUtils.hexToBytes(BroadlinkBindingConstants.BROADLINK_IV), deviceKey, deviceType, logger);
+    }
+
+    protected byte[] decodeDevicePacket(byte[] responseBytes) throws IOException {
+        byte[] rxBytes = BroadlinkProtocol.decodePacket(responseBytes, this.deviceKey,
+                BroadlinkBindingConstants.BROADLINK_IV);
+
+        if (networkTrafficObserver != null) {
+            networkTrafficObserver.onBytesReceived(rxBytes);
+        }
+        return rxBytes;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateItemStatus();
+        }
+    }
+
+    // Can be implemented by devices that should do something on being found; e.g. perform a first status query
+    // protected void onBroadlinkDeviceBecomingReachable() {
+    // updateItemStatus();
+    // }
+
+    // Implemented by devices that can update the openHAB state
+    // model. Return false if something went wrong that requires
+    // a change in the device's online state
+    protected void getStatusFromDevice() throws IOException, BroadlinkException {
+        NetworkUtils.hostAvailabilityCheck(thingConfig.getIpAddress(), 3000, logger);
```

Comment:
This would make the NetworkUtils.hostAvailabilityCheck unnecesairy.
```suggestion
        InetAddress address = InetAddress.getByName(thingConfig.getIpAddress());
        if (!address.isReachable(3000)) {
            throw new BroadlinkHostNotReachableException("Cannot reach " + thingConfig.getIpAddress());
        }
```